qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,532 @@
+    computed before this method is called. The returned Pauli operator must be sorted and simplified.
+
+    Args:
+      `ferm_op`: The fermionic operator to convert.
+      `edge_list`: The qubit-connectivity graph expressed as an edge list.
+
+    Returns:
+      An un-simplified Pauli operator representing `ferm_op`.
+
+    Raises:
+      ValueError: if the type of interaction of any term is unknown.
+    """
+    fer_op_list = ferm_op.to_list()
+    sparse_pauli = None
+    for term in fer_op_list:
+        term_type, facs = _analyze_term(_operator_string(term))
+        if facs[0][1] == "-":  # keep only one of h.c. pair
+            continue
+        ## Following only filters h.c. of some number-excitation op
+        if facs[0][0] == facs[1][0]:  # first op is number op, which is it's own h.c.
+            if len(facs) > 2 and facs[2][1] == "-":  # So, look at next op to skip h.c.
+                continue
+
+        if term_type == TermType.NUMBER:  # a^\dagger_p a_p
+            p = facs[0][0]  # pylint: disable=invalid-name
+            h1_pq = _operator_coefficient(term)
+            sparse_pauli = _add_sparse_pauli(sparse_pauli, _number_operator(edge_list, p, h1_pq))
+            continue
+
+        if term_type == TermType.EXCITATION:
+            (p, q) = [facs[i][0] for i in range(2)]  # p < q always   # pylint: disable=invalid-name
+            h1_pq = _operator_coefficient(term)
+            sparse_pauli = _add_sparse_pauli(
+                sparse_pauli, _excitation_operator(edge_list, p, q, h1_pq)
+            )
+
+        else:
+            facs_reordered, phase = _to_physicist_index_order(facs)
+            h2_pqrs = phase * _operator_coefficient(term)
+            (p, q, r, s) = [facs_reordered[i][0] for i in range(4)]  # pylint: disable=invalid-name
+            if term_type == TermType.DOUBLE_EXCITATION:
+                sparse_pauli = _add_sparse_pauli(
+                    sparse_pauli, _double_excitation(edge_list, p, q, r, s, h2_pqrs)
+                )
+            elif term_type == TermType.COULOMB_EXCHANGE:
+                sparse_pauli = _add_sparse_pauli(
+                    sparse_pauli, _coulomb_exchange(edge_list, p, q, s, h2_pqrs)
+                )
+            elif term_type == TermType.NUMBER_EXCITATION:
+                # Note that h2_pqrs is not divided by 2 here, as in the aqua code
+                sparse_pauli = _add_sparse_pauli(
+                    sparse_pauli, _number_excitation(edge_list, p, q, r, s, h2_pqrs)
+                )
+            else:
+                raise ValueError("Unknown interaction: ", term_type)
+
+    return sparse_pauli
+
+
+def _add_sparse_pauli(qubit_op1: SparsePauliOp, qubit_op2: SparsePauliOp) -> SparsePauliOp:
+    """
+    Return `qubit_op1` + `qubit_op2`, except when either one is `None`.
+    In the latter case, return the one that is not `None`. In other words, assume
+    `None` signifies the additive identity.
```

Comment:
Missing `Args` documentation

Reply:
I fixed this and made many other changes and additions to doc strings to conform to the google style guide as required in *CONTRIBUTING.md*

Reply:
Thanks! I know that most of these are private methods but I am still seeing that some of them now still lack `Returns` documentation.
Maybe we can leave them for now (since most are self-explanatory and private anyways) but we should check inquire about a tool to detect the lack of these docstring aspects (as you also wrote offline).